panic: runtime error: slice bounds out of range

I'm willing to believe it is my usage error, but I'm eliciting a panic from boltdb. This happens in only 3 out of maybe 30 databases that I've created so far and these are 3 of the largest datasets. However, the panics are not related to database size. I'm successfully creating databases as large as 14 GB; the database below is only 3.22 GB when panic sets in.

Each database consists of multiple buckets. One bucket stores all identifier/sequence pairs: the identifiers are approx. 55 bytes each and the sequences are most about 75 bytes though some are over 64 KB. The sequences are also grouped at various levels of granularity. Each grouping file gets its own bucket, named for the file: the keys are each 60-70 bytes and the values are some number of the sequence identifiers concatenated together. Virtually all of the identifiers in the dataset are represented in each grouping file. When the groups are few, the number of identifiers in any one group can be in the millions.

The panic happens during the commit and seems to be related to the size of the values I'm inserting.  Below I'm including the last few debug output lines prior to the panic. The 3 items in each output line are: the actual key, the number of items associated with the key, and the length of the actual value in bytes after concatenating the items into a single value.

While I haven't narrowed it down to the exact byte, my best guess is the limit is 256 MB. The size of the value that causes the panic below is 292,390,837 bytes, yet in the same dataset but different grouping a value of 254,156,848 bytes goes in no problem. I see the 256 MB number used in bolt.node.write as `maxAllocSize`. The documentation says values can go up to 4 GB so maybe I'm on the wrong track.

I grabbed a new version of boltdb just a couple days ago but get the same result.

Currently I'm storing plain ASCII. I hoping to mitigate the problem by using the snappy-go library to compress the values.
```
ACTGATAnelloviridae 112 5757
ATGAGCAnelloviridae 983732 50655530
ATTCCTAnelloviridae 11 563
CAAAAGAnelloviridae 57858 2979600
CAACTAAnelloviridae 1695 87300
CACTCAAnelloviridae 9917 510691
CAGGCGAnelloviridae 144 7415
CATGGCAnelloviridae 7364 379173
CATTTTAnelloviridae 126 6489
CCAACAAnelloviridae 63 3235
CTAGCTAnelloviridae 791 40717
CTCAGAAnelloviridae 225 11589
GGTAGCAnelloviridae 5678213 292390837
panic: runtime error: slice bounds out of range

goroutine 6 [running]:
github.com/boltdb/bolt.(*node).write(0xc297e78070, 0xc367f58000)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/node.go:228 +0x5a5
github.com/boltdb/bolt.(*node).spill(0xc297e78070, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/node.go:364 +0x506
github.com/boltdb/bolt.(*node).spill(0xc297e78000, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/node.go:336 +0x12d
github.com/boltdb/bolt.(*Bucket).spill(0xc342736300, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/bucket.go:535 +0x1c4
github.com/boltdb/bolt.(*Bucket).spill(0xc230488838, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/bucket.go:502 +0xac2
github.com/boltdb/bolt.(*Tx).Commit(0xc230488820, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/tx.go:150 +0x1ee
github.com/boltdb/bolt.(*DB).Update(0xc2080a4000, 0xc208759e70, 0x0, 0x0)
        /home/stryke/surpi_viz/src/github.com/boltdb/bolt/db.go:483 +0x169
go/readdb.(*Db).PutDebug(0xc208034030, 0xc208036410, 0x4e, 0x50, 0xc2745b7360, 0x14, 0x1d, 0xc37cb7e000, 0x56a485, 0x694955)
        /home/stryke/surpi_viz/src/go/readdb/readdb.go:133 +0x71
main.saveIdentifiersDebug(0xc208036410, 0x4e, 0x50, 0xc2d1e30de0, 0xc297e792d0)
        /home/stryke/surpi_viz/src/go/heatmap/heatmap.go:363 +0x2db
created by main.(*table).process
        /home/stryke/surpi_viz/src/go/heatmap/heatmap.go:260 +0x538

goroutine 1 [chan receive]:
main.(*table).process(0xc2080101c0)
        /home/stryke/surpi_viz/src/go/heatmap/heatmap.go:278 +0x893
main.main()
        /home/stryke/surpi_viz/src/go/heatmap/heatmap.go:671 +0x7d0
Command exited with non-zero status 2
482.41user 40.01system 13:06.60elapsed 66%CPU (0avgtext+0avgdata 37673472maxresident)k
168inputs+31538880outputs (0major+3228138minor)pagefaults 0swaps
make: *** [Cameroon_3] Error 2
```

--- a/page.go
+++ b/page.go
@@ -8,7 +8,7 @@ import (
 
 const pageHeaderSize = int(unsafe.Offsetof(((*page)(nil)).ptr))
 
-const maxAllocSize = 0xFFFFFFF
+const maxAllocSize = 0x7FFFFFFF
 const minKeysPerPage = 2
 
 const branchPageElementSize = int(unsafe.Sizeof(branchPageElement{}))

